advanced_source/python_custom_ops.py

```diff
@@ -0,0 +1,254 @@
+# -*- coding: utf-8 -*-
+
+"""
+.. _python-custom-ops-tutorial:
+
+Python Custom Operators
+=======================
+
+.. note::
+   This tutorial is for PyTorch 2.4+ and the PyTorch nightlies.
+
+PyTorch offers a large library of operators that work on Tensors (e.g.
+``torch.add``, ``torch.sum``, etc). However, you might wish to use a new customized
+operator with PyTorch, perhaps written by a third-party library. This tutorial
+shows how to wrap Python functions so that they behave like PyTorch native
+operators. Reasons why you may wish to create a custom operator in PyTorch include:
+
+- Treating an arbitrary Python function as an opaque callable with respect
+  to ``torch.compile`` (that is, prevent ``torch.compile`` from tracing
+  into the function).
+- Adding training support to an arbitrary Python function
+
+Please note that if your operation can be expressed as a composition of
+existing PyTorch operators, then there is usually no need to use the custom operator
+API -- everything (for example ``torch.compile``, training support) should
+just work.
+"""
+######################################################################
+# Example: Wrapping PIL's crop into a custom operator
```

Comment:
Sam as above, add two [cards](https://github.com/pytorch/tutorials/blob/main/beginner_source/template_tutorial.py#L9): what you will learn and prerequisites. Example
What you will learn
* How to create and integrate custom Python functions as PyTorch operators.
* How to optimize with torch.compile and support training with autograd.
* How to add ``FakeTensor`` kernels, register autograd support, and validate operators with ``torch.library.opcheck``.

Prerequisites
* PyTorch 2.4 or later
* Read `torch.compile <link>`__
* Complete `link to relevant tutorial`